library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,565 @@
+require "yast"
+
+require "abstract_method"
+
+module CWM
+  # Represent base for any widget used in CWM. It can be passed as "widget" argument. For more
+  # details about usage see {CWM.ShowAndRun}
+  #
+  # For using widgets design decision is to use subclassing. Reason is to have better separeated
+  # and easily reusable code. Opposite approach is to use instancing of existing classes, but
+  # especially with storing and initializing widgets it can be quite complex. Consider e.g.
+  # instancing of InputField like:
+  #
+  # ```
+  #   widget = InputField.new(
+  #     label: _("My label"),
+  #     help: _("blablbalba" \
+  #       "blablabla" \
+  #       "blablabla"
+  #     ),
+  #     init: Proc.new do |widget|
+  #       ...
+  #     end,
+  #     store: Proc.new do |widget, event|
+  #       ...
+  #     end,
+  #     validate: Proc.new do |widget, event|
+  #       ....
+  #     end
+  #   )
+  # ```
+  #
+  # For this example current subclassing approach looks like
+  # ```
+  #   class MyWidget < CWM::InputField
+  #     def label
+  #       _("My label")
+  #     end
+  #
+  #     def help
+  #       _("blablbalba" \
+  #         "blablabla" \
+  #         "blablabla"
+  #     end
+  #
+  #     def init(widget)
+  #       ...
+  #     end
+  #
+  #     def store(widget, event)
+  #       ...
+  #     end
+  #
+  #     def validate(widget, event)
```

Comment:
We should also update the callback signatures in these examples to (not) include `widget` and `event` according to the final API.